### DISCUSSION TITLE
Custom sorting implementation.

### DISCUSSION THREAD

**Discussion by MrJoy** [avo-hq/avo#783] [Feature requests]
It would be nice if `sort_by` could take a proc/lambda, and when such is provided, use that for applying sorting.

My specific use-cases are two-fold:

1. I have an association (`as: :belongs_to`) I'm showing, and would like to be able to sort by the column that's displayed.
2. I have synthetic fields I'm showing, and would like to be able to apply sorting for the field.

So, in this case, I might go from:

```ruby
field :vendor, as: :belongs_to
field "Era", as: :text, hide_on: [:show] do |model, _resource, _view|
  "#{model.start_year}..#{model.end_year}"
end
```

To something like:

```ruby
field :vendor, as: :belongs_to, sort_by: -> (query, dir) { query.joins(toyline: :vendor).order(name: dir) }
field "Era", as: :text, hide_on: [:show], sort_by: ->(query, dir) { query.order(start_year: dir, end_year: dir) } do |model, _resource, _view|
  "#{model.start_year}..#{model.end_year}"
end
```

**Comment by adrianthedev** [comment]
So that would enable sorting based on computed fields and `belongs_to` associations? That's not such a bad idea!

Would you be open to drafting a PR and championing this idea forward?

**Reply by MrJoy** [reply]
I'll take a stab at it this weekend.

**Reply by adrianthedev** [reply]
Thank you! Let me know if I can help navigate the codebase.

**Reply by MrJoy** [reply]
I've found it refreshingly straightforward so far, so I don't anticipate much need there -- but I do appreciate the offer and will take you up on it if I run into anything particularly thorny/confusing.

**Reply by adrianthedev** [reply]
Haha. That brings some joy to me because I think my code there is a bit tangled up in places. Thanks 🙏

**Reply by MrJoy** [reply]
You're welcome!  Thank you for bringing a fresh approach to a distressingly stagnant niche!
